src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_skadi.cpp

```diff
@@ -15,474 +15,931 @@
-    {468.991f, -516.691f, 104.724f, 0},         //44
-    {461.722f, -517.063f, 104.627f, 0},         //45
-    {455.88f, -517.681f, 104.707f, 0},          //46
-    {450.499f, -519.099f, 104.701f, 0},         //47
-    {444.889f, -518.963f, 104.82f, 0},          //48
-    {440.181f, -518.893f, 104.861f, 0},         //49
-    {434.393f, -518.758f, 104.891f, 0},         //50
-    {429.328f, -518.583f, 104.904f, 0},         //51
-    {423.844f, -518.394f, 105.004f, 0},         //52
-    {418.707f, -518.266f, 105.135f, 0},         //53
-    {413.377f, -518.085f, 105.153f, 0},         //54
-    {407.277f, -517.844f, 104.893f, 0},         //55
-    {401.082f, -517.443f, 104.723f, 0},         //56
-    {394.933f, -514.64f, 104.724f, 0},          //57
-    {388.917f, -514.688f, 104.734f, 0},         //58
-    {383.814f, -515.834f, 104.73f, 0},          //59
-    {377.887f, -518.653f, 104.777f, 0},         //60
-    {371.376f, -518.289f, 104.781f, 0},         //61
-    {365.669f, -517.822f, 104.758f, 0},         //62
-    {359.572f, -517.314f, 104.706f, 0},         //63
-    {353.632f, -517.146f, 104.647f, 0},         //64
-    {347.998f, -517.038f, 104.538f, 0},         //65
-    {341.803f, -516.98f, 104.584f, 0},          //66
-    {335.879f, -516.674f, 104.628f, 0},         //67
-    {329.871f, -515.92f, 104.711f, 0},          //68
-    // Breach Zone
-    {485.4577f, -511.2515f, 115.3011f, 0},      //69
-    {435.1892f, -514.5232f, 118.6719f, 0},      //70
-    {413.9327f, -540.9407f, 138.2614f, 0},      //71
+    EMOTE_BREATH                        = 0,
+    EMOTE_ON_RANGE
 };
 
-enum CombatPhase
+enum Data
 {
-    FLYING,
-    SKADI
+    
+    DATA_LOVE_TO_SKADI                   = 0,
+    FIRST_WAVE_MAX_WARRIORS              = 10,
+    FIRST_WAVE_SIZE                      = 13,
+    GRAUF_PATH_INITIAL                   = 2689300,
+    GRAUF_PATH_RIGHT                     = 2689301,
+    GRAUF_PATH_LEFT                      = 2689302,
+    ACHIEV_LODI_DODI_WE_LOVES_THE_SKADI  = 17726,
 };
 
-enum Spells
+enum Points
 {
-    // Skadi Spells
-    SPELL_CRUSH             = 50234,
-    SPELL_POISONED_SPEAR    = 50225, //isn't being cast
-    SPELL_WHIRLWIND         = 50228, //random target, but not the tank approx. every 20s
-    SPELL_RAPID_FIRE        = 56570,
-    SPELL_HARPOON_DAMAGE    = 56578,
-    SPELL_FREEZING_CLOUD    = 47579,
+    POINT_0                              = 0,
+    POINT_1                              = 1,
+    POINT_9                              = 9,
+    POINT_11                             = 11,
+    POINT_LEFT                           = 21,
+    POINT_RIGHT                          = 22
+};
+
+enum Actions
+{
+    ACTION_START_ENCOUNTER               = 0,
+    ACTION_FLAME,
+    ACTION_GAUNTLET_END,
+    ACTION_HARPOON_HIT,
+};
+
+enum CombatPhase
+{
+    PHASE_FLYING                         = 0,
+    PHASE_GROUND
 };
 
-enum Creatures
+std::list<uint32> SummonSpellsList =
```

Comment:
Yeah, for people that wonder why http://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html ;p